### PR TITLE
feat: semantic cache ui

### PR DESCRIPTION
**Pull request by domeccleston**

### CODE REVIEW

File: apps/dashboard/app/(app)/semantic-cache/analytics/page.tsx [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
+  getSemanticCachesHourly,
+} from "@/lib/tinybird";
+import { redirect } from "next/navigation";
+import { type Interval, IntervalSelect } from "../../apis/[apiId]/select";
 
-interface DataEntry {
+type LogEntry = {
+  hit: number;
+  total: number;
+} & ({ day: string; hour?: never } | { hour: string; day?: never });
```
Comment (chronark) [inline apps/dashboard/app/(app)/semantic-cache/analytics/page.tsx:16]: I'm assuming this one will be just `time` with the new pipes in prod

File: apps/dashboard/app/(app)/semantic-cache/analytics/page.tsx [hunk 2]
```diff
@@ -15,70 +28,152 @@ const tokenCostMap = {
...
+    return redirect("/semantic-cache/new");
+  }
+
+  const { start, end, getSemanticCachesPerInterval } = prepareInterval(interval);
 
-  console.info({ tokenCost, tokens, timeSaved });
+  const query = {
+    start,
+    end,
+    // TODO unhardcode
```
Comment (chronark) [inline apps/dashboard/app/(app)/semantic-cache/analytics/page.tsx:119]:
```suggestion
```

File: apps/dashboard/app/(app)/semantic-cache/analytics/page.tsx [hunk 3]
```diff
@@ -15,70 +28,152 @@ const tokenCostMap = {
...
-    const category = entry.category;
-    if (!dailyCounts[date]) {
-      dailyCounts[date] = {};
-    }
-    if (!dailyCounts[date][category]) {
-      dailyCounts[date][category] = 0;
-    }
-    dailyCounts[date][category] += 1;
-  });
+  const tokens = tokensData.reduce((acc, log) => acc + log.tokens, 0);
```
Comment (chronark) [inline apps/dashboard/app/(app)/semantic-cache/analytics/page.tsx:133]: The new pipes have an `avgTokens` column, but now that I see this, I think maybe a `sumTokens` would be more useful 

wdyt?

File: apps/semantic-cache/src/pkg/streaming.ts [hunk 4]
```diff
@@ -14,7 +14,7 @@ import {
 // import model from "tiktoken/encoders/cl100k_base.json";
 // import { Tiktoken, init } from "tiktoken/lite/init";
 // import wasm from "tiktoken/lite/tiktoken_bg.wasm";
-import { sha256 } from "@unkey/hash";
+import { sha256 } from "../../../../internal/hash/src/index";
```
Comment (chronark) [inline apps/semantic-cache/src/pkg/streaming.ts:17]: huh why?

File: apps/semantic-cache/src/pkg/util/index.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import type { Context } from "@/pkg/hono/app";
-import { sha256 } from "@unkey/hash";
 import type { ChatCompletionMessageParam } from "openai/resources/chat/completions";
+import { sha256 } from "../../../../../internal/hash/src/index";
```
Comment (chronark) [inline apps/semantic-cache/src/pkg/util/index.ts:3]: ?

**Review by chronark — APPROVED** [review]

File: apps/dashboard/app/(app)/semantic-cache/analytics/page.tsx (hunk 3 above)
Comment (domeccleston) [reply apps/dashboard/app/(app)/semantic-cache/analytics/page.tsx:133]: That would be helpful, we should be able to show all tokens cached per gatewayId

File: apps/semantic-cache/src/pkg/streaming.ts (hunk 4 above)
Comment (domeccleston) [reply apps/semantic-cache/src/pkg/streaming.ts:17]: I got a type error here, I don't know why

**Review by chronark — APPROVED** [review]

**Pull request merged**